src/engine/TransitivePathBase.cpp

```diff
@@ -101,13 +101,11 @@ std::shared_ptr<QueryExecutionTree> TransitivePathBase::joinWithIndexScan(
       qec,
       joinWithValues(selectXVariable(ad_utility::makeExecutionTree<IndexScan>(
           qec, Permutation::Enum::SPO,
-          SparqlTriple{TripleComponent{x}, PropertyPath::fromVariable(y),
-                       TripleComponent{z}},
+          SparqlTripleSimple{TripleComponent{x}, y, TripleComponent{z}},
```

Comment:
Make it possible to write `SparqlTripleSimple{x, y, z}` (maybe it already works because the `TripleComponent` constructor is implicit, but please add this to the `SparqlTripleSimple` constructor (the conversion), s.t. the explicit would also work in the future.